### DISCUSSION TITLE
Combine Multiple Steps

### DISCUSSION THREAD

**Discussion by vesubramanian** [community/community#26695] [Actions]
<p>I have a few steps (including actions from market place). I will need to use these same steps in multiple workflows. Is it possible to combine and move them to a separate file (say, action.yml) and refer them in the workflow, so that I don’t have to duplicate the same steps in every workflow?</p>

**Comment by mickeygousset** [comment, marked as answer]
<p>If all the steps are just “run” steps, then you can create a custom composite action that uses those run steps.</p>
<p>However, we don’t support templates right now, so you don’t have the ability to define multiple “uses” steps in a separate YML file.  In that instance, you do have to duplicate the code.</p>

**Comment by laughedelic** [comment]
<p>Isn’t this the same as <a href="https://github.community/t/uses-step-in-composite-action/181289" class="inline-onebox">Uses step in Composite Action</a>? (same  author)</p>

**Comment by vesubramanian** [comment]
<p>Yes <a class="mention" href="/laughedelic">@laughedelic</a> . I posted it to ensure I am conveying the question correctly. Also, the documentation lacks. Hence there is no clarity. I don’t want to duplicate the same steps in multiple workflows. Hence I was looking for a way to make it reusable.</p>

**Comment by jsoref** [comment]
<p>The feature you wanted may have arrived a few months after you asked about it:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/2119212/181099277-65b131ff-2018-46aa-8650-91627c0521a6.png" class="site-icon" width="32" height="32">

      <a href="https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/" target="_blank" rel="noopener">The GitHub Blog</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2022/04/Engineering-Security.png?fit=1200%2C630" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/" target="_blank" rel="noopener">GitHub Actions: Reduce duplication with action composition | GitHub Changelog</a></h3>

  <p>GitHub Actions: Reduce duplication with action composition</p>

  <p>
    <span class="label1">Est. reading time: 1 minute</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
